scripts/create_changelog.py

```diff
@@ -0,0 +1,246 @@
+            'title': title,
+            'creator': creator,
+            'pr_id': pr_id,
+            'authors': authors,
+            'reviewers': reviewers
+        })
+    return pr_info
+
+
+def update_first_contribs(release_data):
+    body = release_data.get('body', '')
+    pattern = r"\* (@\S+) made their first contribution in"
+    matches = re.findall(pattern, body)
+    for match in matches:
+        username = match[1::]
+        first_contribs.add(username)
+
+
+def get_authors(pr_id):
+    url = f"{BASE_URL}/pulls/{pr_id}/commits"
+    authors = []
+    try:
+        res = requests.get(url, headers=HEADERS)
+        res.raise_for_status()
+        commits = res.json()
+        for commit in commits:
+            if commit['commit']['committer']['name'] in AUTOMATED_BOTS:
+                continue
+            if 'author' in commit and 'login' and commit['author']:
+                username = commit['author']['login']
+                if username not in AUTOMATED_BOTS:
+                    authors.append(username)
+                    all_contribs.add(username)
+    except Exception as e:
+        print(f"Failed to fetch commits for PR {pr_id}: {e}")
+    return list(set(authors))
+
+
+def get_reviewers(pr_id, authors):
+    url = f"{BASE_URL}/pulls/{pr_id}/reviews"
+    reviewers = []
+    try:
+        res = requests.get(url, headers=HEADERS)
+        res.raise_for_status()
+        reviews = res.json()
+        for review in reviews:
+            if 'user' in review and 'login' in review['user']:
+                username = review['user']['login']
+                if username not in authors and username not in AUTOMATED_BOTS:
+                    reviewers.append(username)
+                    all_contribs.add(username)
+    except Exception as e:
+        print(f"Failed to fetch reviews for PR {pr_id}: {e}")
+    return list(set(reviewers))
+
+
+def get_latest_tags():
+    url = f"{BASE_URL}/tags?per_page=1000"  # If per_page is not specified then very few tags are fetched
+    try:
+        res = requests.get(url, headers=HEADERS)
+        res.raise_for_status()
+        tags = res.json()
+        tags = [tag['name'] for tag in tags]
+        return tags
+    except Exception as e:
+        print(f"Failed to fetch tags")
+        return None
+
+
+def sort_tags(tag):
+    name = tag.lower()
+    if "beta" in name:
+        return (0, name)  # Beta comes first
+    elif "rc" in name:
+        return (1, name)  # RC comes next
+    else:
+        return (2, name)  # Stable versions come last
+
+
+def save_to_file(filename, ver, date_of_release):
+    with open(filename, 'w') as file:
+        file.write(f"Sage {ver} was released on {date_of_release}. It is available from:\n\n")
+        file.write(f"  * https://www.sagemath.org/download-source.html\n\n")
+        file.write(f"Sage (http://www.sagemath.org/) is developed by volunteers and\n")
+        file.write(f"combines hundreds of open source packages.\n\n")
```

Comment:
It is a slight deviation from the tradition, but it may be better to use `(http://www.sagemath.org)` with no trailing slash for a new start.

Reply:
Done!